controllers/core/configmap_controller.go

```diff
@@ -0,0 +1,104 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/amazon-vpc-resource-controller-k8s/pkg/config"
+	"github.com/aws/amazon-vpc-resource-controller-k8s/pkg/k8s"
+	"github.com/aws/amazon-vpc-resource-controller-k8s/pkg/node/manager"
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// ConfigMapReconciler reconciles a ConfigMap object
+type ConfigMapReconciler struct {
+	client.Client
+	Log         logr.Logger
+	Scheme      *runtime.Scheme
+	NodeManager manager.Manager
+	K8sAPI      k8s.K8sWrapper
+}
+
+//+kubebuilder:rbac:groups=core,resources=configmaps,resourceNames=amazon-vpc-cni,verbs=get;list;watch
+
+// Reconcile handles configmap create/update/delete events by invoking NodeManager
+// to update the status of the nodes as per the enable-windows-ipam flag value.
+
+func (r *ConfigMapReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	logger := r.Log.WithValues("configmap", req.NamespacedName)
+
+	// only update nodes on amazon-vpc-cni updates, return here for other updates
+	if req.Name != config.VpcCniConfigMapName {
+		return ctrl.Result{}, nil
+	}
+	configmap := &corev1.ConfigMap{}
+	if err := r.Client.Get(ctx, req.NamespacedName, configmap); err != nil {
+		if errors.IsNotFound(err) {
+			logger.Info("amazon-vpc-cni configMap is deleted")
```

Comment:
nit: we can add one comment stating, in case map is deleted we need to de register all nodes. This can come later though.